File: growing-shop-java/src/main/java/com/example/growingshop/domain/auth/service/AuthService.java
package com.example.growingshop.domain.auth.service;

import com.example.growingshop.domain.auth.dto.AuthRequest;
import com.example.growingshop.domain.user.domain.User;
import com.example.growingshop.domain.user.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
@RequiredArgsConstructor
public class AuthService {
    private final PasswordEncoder passwordEncoder;
    private final UserRepository userRepository;

    public boolean matchLoginUser(AuthRequest.LoginReq login) {
        Optional<User> user = userRepository.findUsersByLoginId(login.getLoginId());

        return user.filter(value -> passwordEncoder.matches(login.getPassword(), value.getPassword()))
                .isPresent();
    }
}

File: growing-shop-auth/src/main/kotlin/com/example/growingshopauth/auth/service/AuthService.kt
package com.example.growingshopauth.auth.service

import com.example.growingshopauth.auth.dto.AuthRequest
import com.example.growingshopauth.user.repository.ExpandUserRepository
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class AuthService(
    private val passwordEncoder: PasswordEncoder,
    private val userRepository: ExpandUserRepository
) {

    fun matchLoginUser(login: AuthRequest.LoginReq): Boolean =
        userRepository.findUsersByLoginId(login.loginId)
            .filter { passwordEncoder.matches(login.password, it.password) }
            .isPresent
}
